Error building HEAD / master with gecko sdk 2.3

**Describe the bug**
Error building HEAD / master with gecko sdk 2.3

**To Reproduce**
OT Rev: master
HW Platform: EFR32MG12P BRD4161A / Thunderboard sense 2

Build steps
Clone clean OT from gh
```
mkdir -p third_party/silabs/gecko_sdk_suite/
cp -R $SIMPLICITY_STUDIO/developer/sdks/gecko_sdk_suite/v2.3/ third_party/silabs/gecko_sdk_suite/
./bootstrap
make -f examples/Makefile-efr32 BORDER_AGENT=1 BORDER_ROUTER=1 COMMISSIONER=1 UDP_PROXY=1 BOARD=BRD4161A
```
**Expected behavior**
Should be able to build

**Console/log output**
  BUILD    efr32
make -j10 -C build/efr32 --no-print-directory all
  CHECK    .local-version
make --no-print-directory all-recursive
Making all in include
make  all-recursive
Making all in openthread
Making all in platform
make[6]: Nothing to be done for 'all'.
make[6]: Nothing to be done for 'all-am'.
Making all in third_party
Making all in mbedtls
make[4]: Nothing to be done for 'all'.
make[4]: Nothing to be done for 'all-am'.
Making all in src
Making all in core
make[4]: Nothing to be done for 'all'.
Making all in cli
make[4]: Nothing to be done for 'all'.
Making all in ncp
make[4]: Nothing to be done for 'all'.
Making all in diag
make[4]: Nothing to be done for 'all'.
make[4]: Nothing to be done for 'all-am'.
Making all in examples
Making all in platforms
Making all in utils
make[5]: Nothing to be done for 'all'.
Making all in efr32
 cd ../../.. && /bin/bash ./config.status examples/platforms/efr32/Makefile depfiles
config.status: creating examples/platforms/efr32/Makefile
config.status: executing depfiles commands
  CC       libopenthread_efr32_a-radio.o
  CC       libopenthread_efr32_a-system.o
  CC       ../../../third_party/silabs/gecko_sdk_suite/v2.3/hardware/kit/common/bsp/libopenthread_efr32_a-bsp_bcc.o
  CC       ../../../third_party/silabs/gecko_sdk_suite/v2.3/hardware/kit/common/bsp/libopenthread_efr32_a-bsp_init.o
  CC       ../../../third_party/silabs/gecko_sdk_suite/v2.3/platform/Device/SiliconLabs/EFR32MG12P/Source/libopenthread_efr32_a-system_efr32mg12p.o
  CC       ../../../third_party/silabs/gecko_sdk_suite/v2.3/hardware/kit/common/bsp/libopenthread_efr32_a-bsp_stk.o
  CC       ../../../third_party/silabs/gecko_sdk_suite/v2.3/hardware/kit/common/bsp/libopenthread_efr32_a-bsp_stk_leds.o
  CC       ../../../third_party/silabs/gecko_sdk_suite/v2.3/platform/Device/SiliconLabs/EFR32MG12P/Source/GCC/libopenthread_efr32_a-startup_efr32mg12p.o
  CC       ../../../third_party/silabs/gecko_sdk_suite/v2.3/platform/emdrv/gpiointerrupt/src/libopenthread_efr32_a-gpiointerrupt.o
  CC       ../../../third_party/silabs/gecko_sdk_suite/v2.3/platform/emdrv/dmadrv/src/libopenthread_efr32_a-dmadrv.o
  CC       ../../../third_party/silabs/gecko_sdk_suite/v2.3/platform/emdrv/rtcdrv/src/libopenthread_efr32_a-rtcdriver.o
  CC       ../../../third_party/silabs/gecko_sdk_suite/v2.3/platform/emdrv/uartdrv/src/libopenthread_efr32_a-uartdrv.o
In file included from /home/ekarlso/projects/evry/iot-hackathon/openthread/examples/../examples/platforms/efr32/system.c:46:0:
/home/ekarlso/projects/evry/iot-hackathon/openthread/examples/../examples/platforms/efr32/platform-efr32.h:47:8: error: unknown type name 'otInstance'
 extern otInstance *sInstance;
        ^~~~~~~~~~
/home/ekarlso/projects/evry/iot-hackathon/openthread/examples/../examples/platforms/efr32/platform-efr32.h:61:24: error: unknown type name 'otInstance'
 void efr32AlarmProcess(otInstance *aInstance);
                        ^~~~~~~~~~
/home/ekarlso/projects/evry/iot-hackathon/openthread/examples/../examples/platforms/efr32/platform-efr32.h:81:24: error: unknown type name 'otInstance'
 void efr32RadioProcess(otInstance *aInstance);
                        ^~~~~~~~~~
/home/ekarlso/projects/evry/iot-hackathon/openthread/examples/../examples/platforms/efr32/system.c:56:1: error: unknown type name 'otInstance'
 otInstance *sInstance;
 ^~~~~~~~~~
/home/ekarlso/projects/evry/iot-hackathon/openthread/examples/../examples/platforms/efr32/system.c:90:26: error: unknown type name 'otInstance'
 void otSysProcessDrivers(otInstance *aInstance)
                          ^~~~~~~~~~
Makefile:1016: recipe for target 'libopenthread_efr32_a-system.o' failed
make[5]: *** [libopenthread_efr32_a-system.o] Error 1
make[5]: *** Waiting for unfinished jobs....
  CC       ../../../third_party/silabs/gecko_sdk_suite/v2.3/platform/emdrv/ustimer/src/libopenthread_efr32_a-ustimer.o
Makefile:527: recipe for target 'all-recursive' failed
make[4]: *** [all-recursive] Error 1
Makefile:487: recipe for target 'all-recursive' failed
make[3]: *** [all-recursive] Error 1
Makefile:655: recipe for target 'all-recursive' failed
make[2]: *** [all-recursive] Error 1
Makefile:581: recipe for target 'all' failed
make[1]: *** [all] Error 2
examples/Makefile-efr32:272: recipe for target 'do-build-cortex-m4' failed
make: *** [do-build-cortex-m4] Error 2

--- a/examples/platforms/efr32/platform-efr32.h
+++ b/examples/platforms/efr32/platform-efr32.h
@@ -35,8 +35,7 @@
 #ifndef PLATFORM_EFR32_H_
 #define PLATFORM_EFR32_H_
 
-#include <openthread-core-config.h>
-#include <openthread/config.h>
+#include <openthread/instance.h>
 
 #include "em_device.h"
 #include "em_system.h"

